Move Session and Auth providers with auth-related layouts to vue.

farmaazon opened the pull request:
### Pull Request Description

Fixes #13169 

* Session and Auth providers (along with their dependencies) are moved to Vue
* Replace layout hitting their suspense boundaries with vue-router-like solution. Loading screen no longer flicker.
* Because of how vue-router works (its guards have access to globally provided things only), I needed to move our "context stores" to be non-contextual. Used a primitive `createGlobalState` for now, we may discuss other solutions.

### Important Notes

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
~~- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.~~
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [x] Unit tests have been written where possible.
- ~~[ ] If meaningful changes were made to logic or tests affecting Enso Cloud integration in the libraries, 
      or the Snowflake database integration, a run of the [Extra Tests](https://github.com/enso-org/enso/actions/workflows/extra-nightly-tests.yml) has been scheduled.~~
  - If applicable, it is suggested to paste a link to a successful run of the Extra Tests.

farmaazon committed:
WIP

farmaazon committed:
Local storage

farmaazon committed:
Feature flags

farmaazon committed:
Fixes

farmaazon committed:
Add some missing features

farmaazon committed:
Add protected layout and more fixes

farmaazon committed:
Moved test

farmaazon committed:
Finish refactor and cleanup

farmaazon committed:
Some fixes

farmaazon committed:
Remove artificial delay

farmaazon committed:
Fix local storage

farmaazon committed:
Remove debug logs

farmaazon commented on app/gui/src/dashboard/layouts/CloudBrowserDisabled.tsx:
Code moved from `AuthProvider.tsx` without any change.

farmaazon committed:
Self-review

kazcw commented on app/gui/src/authentication/service.ts:
```diff
@@ -118,19 +117,18 @@ export interface AuthService {
 
-  const logger = useLogger()
-  const { router } = useRouter()
+  // TODO[ao] Do we want logger in Vue?
+  // const logger = useLogger()
```
When will this be addressed--should it have an issue?

farmaazon replied:
I decided that console is preferred for now. The only gains of logger I saw before is that we could forbid `console` so all debug logs were caught before merge. But I just restricted `console.trace` and `console.debug`, so I will use these for debugging now.

kazcw commented on app/gui/src/providers/auth.ts:
```diff
@@ -0,0 +1,301 @@
+  /**
+   * Check if given route is allowed for the current user.
+   *
+   * @returns boolean specifying if the route is allowed, or necessary redirect.
+   */
+  function routeGuard(route: RouteLocation) {
```
`true | false | object` is a bit awkward. This seems like a situation for `Result`?

PabloBuchu commented:
so we moved everything to vue or the files? this looks huge

farmaazon commented:
> so we moved everything to vue or the files? this looks huge

The lion share of changed files are just changed imports, because I moved some modules from `dashboard` directory as they are now for wider use.

farmaazon committed:
Review and fixes

farmaazon committed:
Fix session reactivity

farmaazon committed:
Merge branch 'develop' of github.com:enso-org/enso into wip/farmaazon/auth-in-vue

farmaazon committed:
Fix crash when not logged in

PabloBuchu commented:
QA looks ok to me. The only thing I've found is enter to submit is not working on login screen

farmaazon commented:
> QA looks ok to me. The only thing I've found is enter to submit is not working on login screen

Thanks. I reproduced problem with Enter also on develop. Created an issue: https://github.com/enso-org/enso/issues/13246
